src/objectExplorer/objectExplorerService.ts

```diff
@@ -151,18 +152,22 @@ export class ObjectExplorerService {
         };
         this._expandParamsToPromiseMap.set(expandParams, promise);
         const response = await this._connectionManager.client.sendRequest(ExpandRequest.type, expandParams);
-        if (!response) {
+        if (response) {
+            this._currentNode = node;
+            return response;
+        } else {
             this._expandParamsToPromiseMap.delete(expandParams);
+            promise.resolve(undefined);
+            return undefined;
```

Comment:
Either return false or change the return type of this to be `Promise<boolean | undefined>` (returning false seems better)

Reply:
There's no bug here. We are separating database connections from server connections. I'll add a screenshot of 2 nodes.